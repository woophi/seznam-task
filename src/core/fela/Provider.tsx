import { FC } from 'react';
import { RendererProvider as Fela } from 'react-fela';
import { configureFela } from './index';
const felaRenderer = configureFela();

export const FelaProvider: FC = ({ children }) => {
  return <Fela renderer={felaRenderer}>{children}</Fela>;
};
