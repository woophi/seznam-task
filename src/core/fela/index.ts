import plugins from 'fela-preset-web';
import { createRenderer } from 'fela';

export const configureFela = () => {
  const renderer = createRenderer({
    selectorPrefix: 'f_',
    plugins
  });

  return renderer;
};
