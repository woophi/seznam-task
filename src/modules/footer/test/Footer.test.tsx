import { render } from '@testing-library/react';
import { FelaProvider } from 'core/fela/Provider';
import renderer from 'react-test-renderer';
import { Footer } from '../index';

test('Footer renders correctly', () => {
  const component = renderer.create(
    <FelaProvider>
      <Footer />
    </FelaProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Footer has current year', () => {
  const { queryByText } = render(
    <FelaProvider>
      <Footer />
    </FelaProvider>
  );

  const currentYear = new Date().getFullYear();

  expect(queryByText(new RegExp(`${currentYear}`))).toBeTruthy();
});
