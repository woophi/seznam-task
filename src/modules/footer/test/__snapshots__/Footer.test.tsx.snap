// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="f_a f_b f_c f_d"
>
  Â© Copyright 
  2021
</footer>
`;
